tests/test_infer.py

```diff
@@ -20,6 +20,38 @@ def check_uniform_type(in_list):
     )
 
 
+def test_upperlimit_auto(tmpdir, hypotest_args):
```

Comment:
we should add a test checking that the scan returns close to the other upperlimit methods.

Reply:
The grid scan could be tested the same way the auto-scan is tested (with the `pytest.approx` tolerance turned right down because a grid scan needs a large number of points to reach the same precision as the auto-scan).

Reply:
I've added a test comparing the automatic limits to the linear scan. The comparison is done within an absolute tolerance of 0.25, which is the granularity of the scan I compare to.

Reply:
> The comparison is done within an absolute tolerance of 0.25, which is the granularity of the scan I compare to.

Just a quick note that relative tolerances are preferred (c.f. the `help(pytest.approx)` for why).

Reply:
I used an absolute tolerance here because the linear scan is conducted with an absolute step size so naturally the result would have a guaranteed absolute precision rather than a guaranteed relative precision.